compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrDeclarationStorage.kt

```diff
@@ -686,7 +686,8 @@ class Fir2IrDeclarationStorage(
                 IrValueParameterImpl(
                     startOffset, endOffset, origin, symbol,
                     valueParameter.name, index, type,
-                    null, valueParameter.isCrossinline, valueParameter.isNoinline
+                    valueParameter.varargElementType?.toIrType(session, this),
```

Comment:
You could extract varargElementType from `type` by using `org.jetbrains.kotlin.fir.resolve.calls.ArgumentsKt#varargElementType`

Reply:
done